### PR TITLE
fix: improve scoring algorithm accuracy

### PR DESCRIPTION
**Pull request by Danielsio**
## Summary

Addresses findings from the scoring algorithm review. Four fixes implemented, one deferred:

- **P1 -- Fix broken `engineScore` formula**: The old formula `(volume - min) / min` scored a 2.0L engine at 0.25 when minimum was 1.6L. Now treats the minimum as a pass/fail gate: meeting it scores 0.7, with a gentle bonus up to 1.0 for exceeding by 50%. Below minimum = 0.0.
- **P1 -- Km-aware deal score**: `ScoreWithKm` adjusts the deal score based on how the listing's km compares to the cohort's median km. A 30k-km car priced at median now correctly scores higher than a 120k-km car at the same price. Falls back to price-only when km data is unavailable.
- **P2 -- Non-linear year score**: Applied `pow(0.7)` concave curve so newer years cluster near the top (matching real depreciation curves) instead of the old linear scale.
- **P3 -- Junk price filtering**: Prices below 5,000 NIS (placeholder/call-for-price listings) are excluded from cohort median computation.

### Deferred

- **Sub-model cache key**: Requires a schema migration to persist `sub_model` in `listing_history` (currently only in-memory on `RawListing`). Tracked separately.

## Test plan

- [x] All 27 scoring tests pass (including 6 new tests)
- [x] Storage tests pass (sqlite, api)
- [x] `go vet ./...` clean
- [x] `go build ./...` compiles

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **New Features**
  * Deal scoring now accounts for vehicle mileage, improving market comparisons and ranking.

* **Bug Fixes**
  * Excluded invalid/placeholder prices from market median calculations.
  * Adjusted year and engine scoring curves for more realistic assessments and added mileage-based score adjustments with sensible fallbacks.

### CODE REVIEW

File: internal/scoring/scoring.go [hunk 1]
```diff
@@ -261,7 +309,9 @@ func yearScore(year, yearMin, yearMax int) float64 {
 		return 1.0
 	}
 	s := float64(year-yearMin) / float64(yearMax-yearMin)
-	return clamp01(s)
+	// Concave curve: newer years cluster near top, older years spread apart,
+	// matching non-linear depreciation in the Israeli market.
+	return clamp01(math.Pow(s, 0.7))
```
[the remark this answers is not in the document]
Comment (Danielsio) [reply internal/scoring/scoring.go]: Fixed in f7c8923. Clamp the ratio to [0,1] via `clamp01` before passing to `math.Pow`, so out-of-range years (year < yearMin) score 0 instead of producing NaN and being silently skipped. Added test for this edge case.
